process/interceptors/processor/trieNodeInterceptorProcessor.go

```diff
@@ -0,0 +1,48 @@
+package processor
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/trie"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+// TrieNodeInterceptorProcessor is the processor used when intercepting trie nodes
+type TrieNodeInterceptorProcessor struct {
+	interceptedNodes storage.Cacher
+}
+
+// NewTrieNodesInterceptorProcessor creates a new instance of TrieNodeInterceptorProcessor
+func NewTrieNodesInterceptorProcessor(interceptedNodes storage.Cacher) (*TrieNodeInterceptorProcessor, error) {
+	if check.IfNil(interceptedNodes) {
+		return nil, process.ErrNilCacher
+	}
+
+	return &TrieNodeInterceptorProcessor{
+		interceptedNodes: interceptedNodes,
+	}, nil
+}
+
+// Validate checks if the intercepted data can be processed
+func (tnip *TrieNodeInterceptorProcessor) Validate(data process.InterceptedData) error {
```

Comment:
is there nothing to be checked here?

Reply:
The checking is done on the InterceptedData.CheckValidity(), which is called before this method.